service/worker/deployment/deployment_client.go

```diff
@@ -238,6 +249,48 @@ func (d *DeploymentClientImpl) DescribeDeployment(ctx context.Context, namespace
 	}, nil
 }
 
+func (d *DeploymentClientImpl) GetDeploymentReachability(
+	ctx context.Context,
+	namespaceEntry *namespace.Namespace,
+	seriesName string,
+	buildID string,
+) (*workflowservice.GetDeploymentReachabilityResponse, error) {
+	deployInfo, err := d.DescribeDeployment(ctx, namespaceEntry, seriesName, buildID)
+	if err != nil {
+		return nil, err
+	}
+	currentDeployment, err := d.GetCurrentDeployment(ctx, namespaceEntry, seriesName)
+	if err != nil {
+		return nil, err
+	}
+
+	if d.reachabilityCache == nil {
+		cache := newReachabilityCache(
+			metrics.NoopMetricsHandler,
+			d.VisibilityManager,
+			reachabilityCacheOpenWFsTTL,   // TODO (carly) use dc (ie. config.ReachabilityCacheOpenWFsTTL)
+			reachabilityCacheClosedWFsTTL, // TODO (carly) use dc (ie. config.ReachabilityCacheClosedWFsTTL)
+		)
+		d.reachabilityCache = &cache
```

Comment:
this can maybe go in the initialization of the `DeploymentClientImp` right? (inside `deployment/fx.go`)?

Reply:
done! tbh I was intimidated by the fx file, but I think I found the correct initialization function

Reply:
yes it took me...quite some time...to understand how things worked in the fx module and I don't think im done fully comprehending it too - thanks for the fix!